stimpacks/deploy/config.go

```diff
@@ -0,0 +1,352 @@
+			if _, ok := environment.instanceMap[instance.Name]; ok {
+				d.log.Fatal("Error parsing config, duplicate instance name '{}' for environment '{}'", instance.Name, environment.Name)
+			}
+			environment.instanceMap[instance.Name] = j
+
+			// Ensure the instance name does not conflict with the ALL option name.  This is a reserved name for designating a deployment to all instances in an environment via the manual prompt list
+			if strings.ToLower(instance.Name) == strings.ToLower(allOptionPrompt) || strings.ToLower(instance.Name) == strings.ToLower(allOptionCli) {
+				d.log.Fatal("Deployment config cannot have an instance named '{}'. It is a reserved name.", instance.Name)
+			}
+
+			// Create our instance spec if it doesn't exist so we don't have to keep checking if it exists
+			if instance.Spec == nil {
+				instance.Spec = &Spec{}
+			}
+
+			// Merge all of the secrets and environment variables
+			// Instance-level specs take precedence, followed by environment-level then global-level
+			if instance.Spec.Kubernetes.ServiceAccount == "" {
+				if environment.Spec.Kubernetes.ServiceAccount != "" {
+					instance.Spec.Kubernetes.ServiceAccount = environment.Spec.Kubernetes.ServiceAccount
+				} else {
+					instance.Spec.Kubernetes.ServiceAccount = d.config.Global.Spec.Kubernetes.ServiceAccount
+				}
+
+			}
+			if instance.Spec.Kubernetes.Cluster == "" {
+				if environment.Spec.Kubernetes.Cluster != "" {
+					instance.Spec.Kubernetes.Cluster = environment.Spec.Kubernetes.Cluster
+				} else {
+					instance.Spec.Kubernetes.Cluster = d.config.Global.Spec.Kubernetes.Cluster
+				}
+
+			}
+
+			instance.Spec.EnvironmentVars = mergeEnvVars(instance.Spec.EnvironmentVars, environment.Spec.EnvironmentVars, d.config.Global.Spec.EnvironmentVars)
+			instance.Spec.Secrets = mergeSecrets(instance.Spec.Secrets, environment.Spec.Secrets, d.config.Global.Spec.Secrets)
+
+			// Ensure a Kubernetes cluster is set
+			if instance.Spec.Kubernetes.Cluster == "" {
+				d.log.Fatal("Kubernetes cluster is not set for instance '{}' in environment '{}'", instance.Name, environment.Name)
+			}
+
+			// Get Vault details
+			vault := d.stim.Vault()
+			vaultToken, err := vault.GetToken()
+			if err != nil {
+				panic(err)
+			}
+
+			vaultAddress, err := vault.GetAddress()
+			if err != nil {
+				panic(err)
+			}
+
+			// Generate stim env vars
+			stimEnvs := []*EnvironmentVar{}
+
+			stimEnvs = append(stimEnvs, []*EnvironmentVar{
+				&EnvironmentVar{Name: "VAULT_ADDR", Value: vaultAddress},
+				&EnvironmentVar{Name: "VAULT_TOKEN", Value: vaultToken},
+				&EnvironmentVar{Name: "DEPLOY_ENVIRONMENT", Value: environment.Name},
+				&EnvironmentVar{Name: "DEPLOY_INSTANCE", Value: instance.Name},
+				&EnvironmentVar{Name: "DEPLOY_CLUSTER", Value: instance.Spec.Kubernetes.Cluster},
+			}...)
+
+			// Generate the Kube config secret
+			var stimSecrets []*v2e.SecretItem
+			if instance.Spec.Kubernetes.ServiceAccount != "" {
```

Comment:
This if and else can be removed since we would have called Fatal farther up (assuming the comment on line 179 was done)